packages/server/postgres/migrations/1635354572171_addIntegrationProviderAndToken.ts

```diff
@@ -0,0 +1,91 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+  DO $$
+  BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProvidersEnum') THEN
+      CREATE TYPE "IntegrationProvidersEnum" AS ENUM (
+        'gitlab',
+        'mattermost'
+      );
+    END IF;
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProviderTypesEnum') THEN
+      CREATE TYPE "IntegrationProviderTypesEnum" AS ENUM (
+        'pat',
+        'oauth2',
+        'webhook'
+      );
+    END IF;
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProviderScopesEnum') THEN
+      CREATE TYPE "IntegrationProviderScopesEnum" AS ENUM (
+        'global',
+        'org',
+        'team'
+      );
+    END IF;
+    CREATE TABLE IF NOT EXISTS "IntegrationProvider" (
+      "id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+      "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+      "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+      "provider" "IntegrationProvidersEnum" NOT NULL,
+      "type" "IntegrationProviderTypesEnum" NOT NULL,
+      "scope" "IntegrationProviderScopesEnum" NOT NULL,
+      "scopeGlobal" BOOLEAN GENERATED ALWAYS AS (
+        CASE
+          WHEN "scope" = 'global' THEN TRUE
+          ELSE NULL
+        END
+      ) STORED,
+      "orgId" VARCHAR(100),
+      "teamId" VARCHAR(100),
+      "isActive" BOOLEAN DEFAULT TRUE NOT NULL,
+      "name" VARCHAR(250) NOT NULL,
+      "providerMetadata" JSONB NOT NULL DEFAULT '{}',
```

Comment:
i think that was chosen to differentiate between tokenMetadata & providerMetadata (it gets confusing when you're in the thick of it, especially because the types weren't originally very strict).

IMO i think we should just use columns, then we get types for free!